owncloudComLibrary/src/androidTest/java/com/owncloud/android/lib/GetRemoteStatusOperationTest.kt

```diff
@@ -0,0 +1,118 @@
+package com.owncloud.android.lib
+
+import android.net.Uri
+import com.owncloud.android.lib.resources.status.GetRemoteStatusOperation
+import com.owncloud.android.lib.resources.status.HttpScheme.HTTPS_PREFIX
+import com.owncloud.android.lib.resources.status.HttpScheme.HTTP_PREFIX
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class GetRemoteStatusOperationTest {
+
+    @Test
+    fun urlStartingWithHttpMustBeDetectedAsSuch() {
+        assertTrue(GetRemoteStatusOperation.usesHttpOrHttps(Uri.parse(HTTP_SOME_OWNCLOUD)))
+    }
+
+    @Test
+    fun urlStartingWithHttpsMustBeDetectedAsSuch() {
+        assertTrue(GetRemoteStatusOperation.usesHttpOrHttps(Uri.parse(HTTPS_SOME_OWNCLOUD)))
+    }
+
+    @Test
+    fun incompleteUrlWithoutHttpsOrHttpSchemeMustBeDetectedAsSuch() {
+        assertFalse(GetRemoteStatusOperation.usesHttpOrHttps(Uri.parse(SOME_OWNCLOUD)))
+    }
+
+    @Test
+    fun completeUrlWithHttpMustBeReturnedAsSuch() {
+        assertEquals(
+            Uri.parse(HTTP_SOME_OWNCLOUD),
+            GetRemoteStatusOperation.buildFullHttpsUrl(Uri.parse(HTTP_SOME_OWNCLOUD))
+        )
+    }
+
+    @Test
+    fun completeUrlWithHttpsMustBeReturnedAsSuch() {
+        assertEquals(
+            Uri.parse(HTTPS_SOME_OWNCLOUD),
+            GetRemoteStatusOperation.buildFullHttpsUrl(Uri.parse(HTTPS_SOME_OWNCLOUD))
+        )
+    }
+
+    @Test
+    fun incompleteUrlWithoutHttpPrefixMustBeConvertedToProperUrlWithHttpsPrefix() {
+        assertEquals(
+            Uri.parse(HTTPS_SOME_OWNCLOUD),
+            GetRemoteStatusOperation.buildFullHttpsUrl(Uri.parse(SOME_OWNCLOUD))
+        )
+    }
+
+    @Test
+    fun completeUrlWithSubdirAndHttpsMustBeReturnedAsSuch() {
+        assertEquals(
+            Uri.parse(HTTPS_SOME_OWNCLOUD_WITH_SUBDIR), GetRemoteStatusOperation.buildFullHttpsUrl(
+                Uri.parse(
+                    HTTPS_SOME_OWNCLOUD_WITH_SUBDIR
+                )
+            )
+        )
+    }
+
+    @Test
+    fun incompleteUrlWithSubdirAndWithoutHttpPrefixMustBeConvertedToProperUrlWithHttpsPrefix() {
+        assertEquals(
+            Uri.parse(HTTPS_SOME_OWNCLOUD_WITH_SUBDIR), GetRemoteStatusOperation.buildFullHttpsUrl(
+                Uri.parse(
+                    SOME_OWNCLOUD_WITH_SUBDIR
+                )
+            )
+        )
+    }
+
+    @Test
+    fun ipMustBeConvertedToProperUrl() {
```

Comment:
I would improve the test naming, something like this:

`build_full_https_url_ok_ip`
`build_full_https_url_ok_ip_with_preffix`
`build_full_https_url_ok_ip_with_port`

And so on, what do you think?

Reply:
done